Path: bbn.com!bbn!husc6!think!ames!elroy!devvax!lroot
From: lroot@devvax.JPL.NASA.GOV (The Superuser)
Newsgroups: comp.sources.bugs
Subject: perl 1.0 patch #5
Summary: This is an official patch for perl 1.0.  Please apply it.
Message-ID: <1155@devvax.JPL.NASA.GOV>
Date: 26 Jan 88 18:00:14 GMT
Organization: Jet Propulsion Laboratory, Pasadena, CA
Lines: 88

System: perl version 1.0
Patch #: 5
Priority: MEDIUM for Sys V, LOW otherwise
Subject: a2p didn't make use of the config.h generated by Configure
From: "Arnold D. Robbins" <arnold@EMORYU2.ARPA>

Description:
	The a2p program used index() and bcopy(), both of do not exist
	everywhere.  Since Configure was already figuring out about those
	functions, it is fairly trivial to get a2p to make use of the info.

Fix:	From rn, say "| patch -p0 -d DIR", where DIR is your perl source
			      ^^^
	directory.  Outside of rn, say "cd DIR; patch -p0 <thisarticle".
	If you don't have the patch program, apply the following by hand,
	or get patch.

>>>	*** NOTE: the -p0 is necessary above or patch won't find a2p.h ***

	If patch indicates that patchlevel is the wrong version, you may need
	to apply one or more previous patches, or the patch may already
	have been applied.  See the patchlevel.h file to find out what has or
	has not been applied.  In any event, don't continue with the patch.

	If you are missing previous patches they can be obtained from me:

	Larry Wall
	lwall@jpl-devvax.jpl.nasa.gov

	If you send a mail message of the following form it will greatly speed
	processing:

	Subject: Command
	@SH mailpatch PATH perl 1.0 LIST
		   ^ note the c

	where PATH is a return path FROM ME TO YOU in Internet notation, and
	LIST is the number of one or more patches you need, separated by spaces,
	commas, and/or hyphens.  Saying 35- says everything from 35 to the end.

	You can also get the patches via anonymous FTP from
	jpl-devvax.jpl.nasa.gov (128.149.8.43).

Index: patchlevel.h
Prereq: 4
1c1
< #define PATCHLEVEL 4
---
> #define PATCHLEVEL 5
 
Index: x2p/a2p.h
Prereq: 1.0
*** x2p/a2p.h.old	Tue Jan 26 09:53:22 1988
--- x2p/a2p.h	Tue Jan 26 09:53:22 1988
***************
*** 1,10 ****
! /* $Header: a2p.h,v 1.0 87/12/18 13:06:58 root Exp $
   *
   * $Log:	a2p.h,v $
   * Revision 1.0  87/12/18  13:06:58  root
   * Initial revision
   * 
   */
  
  #include "handy.h"
  #define Nullop 0
--- 1,21 ----
! /* $Header: a2p.h,v 1.0.1.1 88/01/26 09:52:30 root Exp $
   *
   * $Log:	a2p.h,v $
+  * Revision 1.0.1.1  88/01/26  09:52:30  root
+  * patch 5: a2p didn't use config.h.
+  * 
   * Revision 1.0  87/12/18  13:06:58  root
   * Initial revision
   * 
   */
+ 
+ #define VOIDUSED 1
+ #include "../config.h"
+ 
+ #ifndef BCOPY
+ #   define bcopy(s1,s2,l) memcpy(s2,s1,l);
+ #   define bzero(s,l) memset(s,0,l);
+ #endif
  
  #include "handy.h"
  #define Nullop 0
